src/she/CMakeLists.txt

```diff
@@ -210,3 +219,9 @@ if(USE_SKIA_BACKEND)
   target_link_libraries(she
     ${SKIA_LIBRARIES})
 endif()
+
+if(WITH_GTK_FILE_DIALOG_SUPPORT)
+  target_link_libraries(she
+    ${GTKMM_LIBRARIES}
+    app-lib)

```

Comment:
This is circular dependency: `app-lib` depends on `she`, so `she` cannot depend on `app-lib`. In case we need the thumbnail generator in the `she` side, we should think about a interface that can provide `she` and must be implemented by the `app` side to generate thumbnails.